packages/gateway/src/gateway.js

```diff
@@ -147,6 +150,23 @@ async function _gatewayFetch(
   return gwResponse
 }
 
+/**
+ * @param {string} gwUrl
+ * @param {Request} request
+ */
+function _getHeadersForGateway(gwUrl, request) {
+  if (gwUrl !== 'https://ipfs.io') {
+    return {}
+  }
+
+  const existingProxies = request.headers['X-Forwarded-For']
+    ? `, ${request.headers['X-Forwarded-For']}`
+    : ''
+  return {
+    'X-Forwarded-For': `${request.headers['cf-connecting-ip']}${existingProxies}`,
```

Comment:
Shouldn't this have a delimiter?

Reply:
The delimeter is set above within `existingProxies` var

Reply:
Oh damn, yes, I missed that!